ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -160,34 +156,28 @@ private[dao] final class TransactionsReader(
             ),
         )
       )
-      .map { x =>
-        EventsRange(
-          startExclusive = (Offset.beforeBegin, 0),
-          endInclusive = (activeAt, x.endInclusive),
-        )
-      }
 
   private def getEventSeqIdRange(
       startExclusive: Offset,
       endInclusive: Offset,
-  )(implicit loggingContext: LoggingContext): Future[EventsRange[(Offset, Long)]] =
+  )(implicit loggingContext: LoggingContext): Future[EventsRange] =
     dispatcher
       .executeSql(dbMetrics.getEventSeqIdRange)(implicit connection =>
         queryNonPruned.executeSql(
-          eventSeqIdReader.readEventSeqIdRange(EventsRange(startExclusive, endInclusive))(
-            connection
+          EventsRange(
+            startExclusiveOffset = startExclusive,
+            startExclusiveEventSeqId = eventStorageBackend.maxEventSequentialIdOfAnObservableEvent(
+              startExclusive
+            )(connection),
+            endInclusiveOffset = endInclusive,
+            endInclusiveEventSeqId =
+              eventStorageBackend.maxEventSequentialIdOfAnObservableEvent(endInclusive)(connection),
```

Comment:
You're calling `maxEventSequentialIdOfAnObservableEvent` twice here, and they access the ledger end cache which might change between the calls. I tried to analyze that it seems it's sound but non trivial

Reply:
I don't think it matters much, because the ledger end is only used as a proxy to deduce the fallback seq id, and if the ledger end moves, then we will have meta entries, which from we will deduce the same seq id again.

Reply:
we are calling it in the right order, por eso es bueno